File: frontend/lib/model/AuthCredential.dart
enum UserRole {
  PROJECT_PROPOSER,
  EXPERT,
  DESIGNER_ENTITY,
  DESIGNER_PERSON,
  NOT_COMPLETED
}

class AuthCredential {
  String _mail;
  String _password;
  List<UserRole> _roles = [];

  AuthCredential(this._mail, this._password);

  AuthCredential.complete(this._mail, this._password, this._roles);

  String getMail() {
    return _mail;
  }

  String getPassword() {
    return _password;
  }

  void setPassword(String pass) {
    this._password = pass;
  }

  void setRoles(List<UserRole> roles) {
    this._roles = roles;
  }

  List<UserRole> getRoles() {
    return _roles;
  }

  List<String> getRolesToString() {
    List<String> roles = [];
    for (UserRole item in _roles) {
      roles.add(item.toString().substring(item.toString().indexOf('.') + 1));
    }
    return roles;
  }
}

File: backend/src/main/java/it/unicam/qwert123/doit/backend/models/AuthCredential.java
package it.unicam.qwert123.doit.backend.models;

import java.util.ArrayList;
import java.util.Collection;
import java.util.List;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.SimpleGrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@SuppressWarnings("serial")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
@Document(collection = "authcredential")
public class AuthCredential implements UserDetails {

    @EqualsAndHashCode.Include
    @Id
    private String mail;
    private String password;
    private List<Role> roles;

    public enum Role  {
        ADMIN, DESIGNER, EXPERT, NOT_COMPLETED, PROJECT_PROPOSER
    }

    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        List<GrantedAuthority> auth = new ArrayList<GrantedAuthority>();
        for (Role role : roles)
            auth.add(new SimpleGrantedAuthority(role.name()));
        return auth;
    }

    @Override
    public String getPassword() {
        return password;
    }

    @Override
    public String getUsername() {
        return mail;
    }

    @Override
    public boolean isAccountNonExpired() {
        return true;
    }

    @Override
    public boolean isAccountNonLocked() {
        return true;
    }

    @Override
    public boolean isCredentialsNonExpired() {
        return true;
    }

    @Override
    public boolean isEnabled() {
        return true;
    }

}
